src/resolvers.ts

```diff
@@ -1,63 +1,82 @@
 import { makeExecutableSchema } from '@graphql-tools/schema';
-import { loadSchemaSync } from '@graphql-tools/load';
-import { GraphQLFileLoader } from '@graphql-tools/graphql-file-loader';
-
 import { Resolvers } from './resolvers-types.js';
+import { visit, print, parse } from 'graphql';
+import fs from 'fs';
 import {
   TracingState,
   setSpanNameFromGraphQLContext,
 } from './tracing/tracer.js';
 
-export { resolvers, schema };
-
-const resolvers: Resolvers = {
+const fullResolvers: Resolvers = {
   Query: {
     events: async (_, { input }, context) => {
-      const graphQLSpan = setSpanNameFromGraphQLContext(
-        context,
-        'events.graphql'
-      );
-
+      const graphQLSpan = setSpanNameFromGraphQLContext(context, 'events.graphql');
       return context.db_client.getEvents(input, {
         tracingState: new TracingState(graphQLSpan),
       });
     },
-
     actions: async (_, { input }, context) => {
-      const graphQLSpan = setSpanNameFromGraphQLContext(
-        context,
-        'actions.graphql'
-      );
+      const graphQLSpan = setSpanNameFromGraphQLContext(context, 'actions.graphql');
       return context.db_client.getActions(input, {
         tracingState: new TracingState(graphQLSpan),
       });
     },
-
     networkState: async (_, __, context) => {
-      const graphQLSpan = setSpanNameFromGraphQLContext(
-        context,
-        'networkState.graphql'
-      );
+      const graphQLSpan = setSpanNameFromGraphQLContext(context, 'networkState.graphql');
       return context.db_client.getNetworkState({
         tracingState: new TracingState(graphQLSpan),
       });
     },
-
     blocks: async (_, { query, limit, sortBy }, context) => {
-      const graphQLSpan = setSpanNameFromGraphQLContext(
-        context,
-        'blocks.graphql'
-      );
+      const graphQLSpan = setSpanNameFromGraphQLContext(context, 'blocks.graphql');
       return context.db_client.getBlocks(query, limit, sortBy, {
         tracingState: new TracingState(graphQLSpan),
       });
     },
   },
 };
 
+let resolvers: Resolvers = fullResolvers;
+let typeDefs: string | undefined = undefined;
+
+// If the ENABLED_QUERIES environment variable is set, filter the schema and resolvers.
+if (process.env.ENABLED_QUERIES !== undefined) {
+  const enabledQueries = process.env.ENABLED_QUERIES.split(',').map((q) => q.trim());
```

Comment:
Could be a follow-up PR, but it would be good to document what the valid options are for this delimited list.